packages/main/src/backend/index.ts

```diff
@@ -4,42 +4,71 @@ import { scrapeFinancialAccountsAndFetchTransactions } from '@/backend/import/im
 import moment from 'moment';
 import * as configManager from './configManager/configManager';
 import * as Events from './eventEmitters/EventEmitter';
+import { EventNames } from './eventEmitters/EventEmitter';
 import outputVendors from './export/outputVendors';
-import * as bankScraper from './import/bankScraper';
 import logger from '../logging/logger';
 
 export { CompanyTypes } from 'israeli-bank-scrapers-core';
 export { Events, configManager, outputVendors };
 
-export const { inputVendors } = bankScraper;
+let intervalId: NodeJS.Timeout | null = null;
+
+export async function scrapePeriodicallyIfNeeded(config: Config, optionalEventPublisher?: Events.EventPublisher) {
+  const hoursInterval = config.scraping.periodicScrapingIntervalHours;
+  optionalEventPublisher = optionalEventPublisher ?? new Events.BudgetTrackingEventEmitter();
+
+  stopPeriodicScraping();
+
+  if (hoursInterval) {
+    await optionalEventPublisher.emit(EventNames.LOG, {
+      message: `Setting up periodic scraping every ${hoursInterval} hours`,
+    });
+    intervalId = setInterval(
+      async () => {
+        await scrapeAndUpdateOutputVendors(config, optionalEventPublisher);
+      },
+      hoursInterval * 1000 * 60 * 60,
+    );
+  }
+}
+
+export function stopPeriodicScraping() {
+  if (intervalId) {
+    clearInterval(intervalId);
+  }
+}
 
 export async function scrapeAndUpdateOutputVendors(config: Config, optionalEventPublisher?: Events.EventPublisher) {
   const eventPublisher = optionalEventPublisher ?? new Events.BudgetTrackingEventEmitter();
+  await eventPublisher.emit(EventNames.LOG, {
```

Comment:
@baruchiro this is a really strange one. It seems that the first event published gets lost. I spent an hour trying to understand why the IMPORT_PROCESS_START doesn't do anything until I added an event before it, and only then it started to work. Couldn't figure out why